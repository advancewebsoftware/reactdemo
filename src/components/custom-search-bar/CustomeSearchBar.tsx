import React from 'react';
import {Colors} from '../../theme';
import {Images} from '../../assets';
import {styles} from './CustomeSearchBarStyle';
import {homeScreenString} from '../../constants';
import {Image, TextInput, TouchableOpacity, View} from 'react-native';

interface CustomeSearchBarProps {
  onCancelPress: () => void;
  onChangeText?: ((text: string) => void) | undefined;
}

const CustomeSearchBar = ({
  onCancelPress,
  onChangeText,
}: CustomeSearchBarProps) => {
  return (
    <View style={styles.container}>
      <Image source={Images.search_ic} style={styles.image} />
      <TextInput
        autoCorrect={false}
        style={styles.inputStyle}
        onChangeText={onChangeText}
        placeholder={homeScreenString.search}
        placeholderTextColor={Colors.placeholder}
      />
      <TouchableOpacity onPress={onCancelPress}>
        <Image source={Images.cancel_ic} style={styles.image} />
      </TouchableOpacity>
    </View>
  );
};

export default CustomeSearchBar;
