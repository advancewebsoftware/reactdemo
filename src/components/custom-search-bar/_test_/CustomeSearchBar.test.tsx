import React from 'react';
import {render} from '@testing-library/react-native';
import CustomeSearchBar from '../CustomeSearchBar';

const component = <CustomeSearchBar onCancelPress={()=>{}} />;

describe('CustomeSearchBar', () => {
  it('renders correctly', () => {
    const {toJSON} = render(component);
    expect(toJSON()).toMatchSnapshot();
  });
});
