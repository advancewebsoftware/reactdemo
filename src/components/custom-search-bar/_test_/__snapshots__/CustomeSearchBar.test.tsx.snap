// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomeSearchBar renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 148.5,
      "flexDirection": "row",
      "gap": 15,
      "height": 73.92857142857143,
      "marginBottom": 32.857142857142854,
      "marginHorizontal": 40,
      "marginTop": 32.857142857142854,
      "paddingHorizontal": 32,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/assets/images/search.png",
      }
    }
    style={
      {
        "height": 32.857142857142854,
        "tintColor": "#000000",
        "width": 40,
      }
    }
  />
  <TextInput
    autoCorrect={false}
    placeholder="Search"
    placeholderTextColor="#232323"
    style={
      {
        "flex": 1,
        "fontSize": 21,
        "fontWeight": "400",
        "height": 73.92857142857143,
      }
    }
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/images/cancel.png",
        }
      }
      style={
        {
          "height": 32.857142857142854,
          "tintColor": "#000000",
          "width": 40,
        }
      }
    />
  </View>
</View>
`;
